lib/widgets/flame_animation_widget.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flutter/material.dart' hide Animation;
+import 'package:flame/animation.dart';
+
+import './flame_sprite_widget.dart';
+
+import 'dart:math';
+
+class FlameAnimationWidget extends StatefulWidget {
+  final Animation animation;
+  final bool center;
+  final bool play;
+
+  FlameAnimationWidget({
+    this.animation,
+    this.play = true,
+    this.center = false,
+  }) : assert(animation.loaded(), 'Animation must be loaded');
+
+  @override
+  State createState() => _FlameAnimationWidget();
+}
+
+class _FlameAnimationWidget extends State<FlameAnimationWidget>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  double _lastUpdated;
+
+  @override
+  void didUpdateWidget(oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.play) {
+      _initAnimation();
+    } else {
+      _pauseAnimation();
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    _controller = AnimationController(vsync: this)
+      ..addListener(() {
+        final now = DateTime.now().millisecond.toDouble();
+
+        final dt = max(0, (now - _lastUpdated) / 1000).toDouble();
+        widget.animation.update(dt);
+
+        setState(() {
+          _lastUpdated = now;
+        });
+      });
+
+    widget.animation.onCompleteAnimation = _pauseAnimation;
+
+    if (widget.play) {
+      _initAnimation();
+    }
+  }
+
+  void _initAnimation() {
+    setState(() {
+      widget.animation.reset();
+      _lastUpdated = DateTime.now().millisecond.toDouble();
+      _controller.repeat(
+          // -/+ 60 fps
```

Comment:
-/+? what does that mean? can't have negative fps haha

Reply:
More or less? So-so, about, etc...

Reply:
ohh haha you can use `Approximately 60 fps`.